### PR TITLE
[BigQuery] Missed one spot for aliasing

### PR DESCRIPTION
**Pull request by Tang8330**
As per title

### CODE REVIEW

**Review by nathan-artie — APPROVED** [review]

File: lib/kafkalib/partition/settings.go [hunk 1]
```diff
@@ -43,7 +44,7 @@ func (b *BigQuerySettings) GenerateMergeString(values []string) (string, error)
 	case "time":
 		switch b.PartitionBy {
 		case "daily":
-			return fmt.Sprintf(`DATE(c.%s) IN (%s)`, b.PartitionField, array.StringsJoinAddSingleQuotes(values)), nil
+			return fmt.Sprintf(`DATE(%s.%s) IN (%s)`, constants.TargetAlias, b.PartitionField, array.StringsJoinAddSingleQuotes(values)), nil
```
Comment (nathan-artie) [inline lib/kafkalib/partition/settings.go:47]: Sneaky!

Comment (nathan-artie) [reply lib/kafkalib/partition/settings.go:47]: Is `b.PartitionField` always a column? Should we be quoting it?

Comment (nathan-artie) [reply lib/kafkalib/partition/settings.go:47]: If so could do `QuoteTableAliasColumns(constants.TargetAlias, columns.Column(b.PartitionField, typing.Invalid))`

Comment (nathan-artie) [reply lib/kafkalib/partition/settings.go:47]: I think using the `sql` package at all may cause an import cycle, so maybe something to look at after this PR.

Comment (Tang8330) [reply lib/kafkalib/partition/settings.go:47]: Yes, we should escape it

Comment (Tang8330) [reply lib/kafkalib/partition/settings.go:47]: I'll leave a TODO

Comment (Tang8330) [reply lib/kafkalib/partition/settings.go:47]: Actually, I can just move this to the bq pkg

File: clients/bigquery/merge.go [hunk 2]
```diff
@@ -49,3 +53,29 @@ func (s *Store) Merge(tableData *optimization.TableData) error {
 		RetryColBackfill: true,
 	})
 }
+
+func generateMergeString(bqSettings *partition.BigQuerySettings, dialect sql.Dialect, values []string) (string, error) {
```
Comment (nathan-artie) [inline clients/bigquery/merge.go:57]: Nice.

**Review by nathan-artie — APPROVED** [review]

**Pull request merged**